src/main/java/com/cburch/logisim/std/io/extra/Slider.java

```diff
@@ -65,50 +64,50 @@
    */
   public static final String _ID = "Slider";
+  
+  private static final int MAXIMUM_NUMBER_OF_BITS = 8;
+  private static final int MAXIMUM_SLIDER_POSITION = (1 << MAXIMUM_NUMBER_OF_BITS) - 1;
 
   public static class Poker extends InstancePoker {
-    private SliderValue data;
     private boolean dragging = false;
-    private BitWidth b;
 
     @Override
     public void mouseDragged(InstanceState state, MouseEvent e) {
       if (dragging) {
-        byte sliderPosition = (byte) (e.getX() - state.getInstance().getBounds().getX() - 10);
-        if (sliderPosition < 0) sliderPosition = 0;
-        else if (sliderPosition > SliderWidth) sliderPosition = SliderWidth;
-        if (data.right_to_left) sliderPosition = (byte) (SliderWidth - sliderPosition);
-        long value =
-            Math.round(sliderPosition * (Math.pow(2, b.getWidth()) - 1) / SliderWidth);
-        data.setCurrentValue(value);
-        data.setCurrentX(sliderPosition);
-        state.getAttributeSet().setValue(ATTR_VALUE, value);
+        var data = (SliderValue) state.getData();
+        if (data == null) {
+          data = new SliderValue();
+          data.setDirection(state.getAttributeValue(ATTR_DIR) == RIGHT_TO_LEFT);
+          data.setCurrentBitWidth(state.getAttributeValue(WIDTH).getWidth());
+          state.setData(data);
+        }
+        data.setSliderPosition(e.getX() - state.getInstance().getBounds().getX() - 10);
         state.fireInvalidated();
       }
     }
 
     @Override
     public void mousePressed(InstanceState state, MouseEvent e) {
-      this.data = getValueState(state);
-      this.b = state.getAttributeValue(StdAttr.WIDTH);
-      Bounds bds = state.getInstance().getBounds();
-      Rectangle slider =
-          new Rectangle(
-              bds.getX() + data.getCurrentX() + 5, bds.getY() + bds.getHeight() - 16, 12, 12);
+      final var data = (SliderValue) state.getData();
+      final var sliderPosition = (data != null) ? data.getSliderPosition() : 
+          state.getAttributeValue(ATTR_DIR) == RIGHT_TO_LEFT ? MAXIMUM_SLIDER_POSITION : 0;
+      final var bounds = state.getInstance().getBounds();
+      final var slider = new Rectangle(bounds.getX() + sliderPosition + 5, 
+          bounds.getY() + bounds.getHeight() - 16, 12, 12);
       // check if clicking slider rectangle
-      if (slider.contains(e.getX(), e.getY())) this.dragging = true;
+      dragging = slider.contains(e.getX(), e.getY());
     }
 
     @Override
     public void mouseReleased(InstanceState state, MouseEvent e) {
-      this.dragging = false;
+      dragging = false;
     }
   }
 
   public static class SliderValue implements InstanceData, Cloneable {
-    private long currentvalue = 0;
-    private byte bitwidth = 8, currentx = 0;
-    private boolean right_to_left = false;
+    private int nrOfBits = MAXIMUM_NUMBER_OF_BITS;
+    private int sliderPosition = 0;
+    private boolean rightToLeft = false;
```

Comment:
`var` instead of `boolean` please.

Reply:
Sorry, but as these are global class variables, it won't accept var. I could make an initialiser where they are assigned, but that would provide more LOC

Reply:
Ok, no need to touch then.